lib/sensible_logging.rb

```diff
@@ -20,21 +20,43 @@ module Sinatra
   # Sensible logging library for Sinatra based Apps
   module SensibleLogging
     def sensible_logging(
-      logger: Logger.new(STDOUT),
-      log_tags: [],
-      use_default_log_tags: true,
-      exclude_params: [],
-      tld_length: 1
+      opts
     )
-      use RequestId
-      use TaggedLogger, logger, log_tags, use_default_log_tags, tld_length
-      use RequestLogger, exclude_params
+      options = default_options(opts)
+
+      setup_middlewares(options)
 
       before do
         env['rack.errors'] = env['rack.logger'] = env['logger']
         logger.level = settings.log_level unless settings.log_level.nil?
       end
     end
+
+    private
+
+    def setup_middlewares(options)
```

Comment:
while I like this has been split out, I'm still not keen on using a lose signature that just accepts an unstructured object.

that being said, it's a private method, so if the options are defined publicly, I'm not too fussed about this